app/src/main/java/com/cornellappdev/android/eateryblue/ui/viewmodels/LoginViewModel.kt

```diff
@@ -29,42 +175,50 @@ class LoginViewModel @Inject constructor(
         getSavedLoginInfo()
     }
 
-    fun getSavedLoginInfo() = viewModelScope.launch {
-        isLoggedIn = if (userPreferencesRepository.getIsLoggedIn()) {
+    private fun getSavedLoginInfo() = viewModelScope.launch {
+        if (userPreferencesRepository.getIsLoggedIn()) {
             val loginInfo = userPreferencesRepository.fetchLoginInfo()
-            LoggedInStatus.IsLoggedIn(loginInfo.first, loginInfo.second)
-        } else {
-            LoggedInStatus.NotLoggedIn
+            getUser(loginInfo.first)
         }
     }
 
-    fun saveLoginInfo(username: String, password: String) = viewModelScope.launch {
-        userPreferencesRepository.saveLoginInfo(username, password)
-        userPreferencesRepository.setIsLoggedIn(true)
-    }
-
     fun getUser(sessionId: String) = viewModelScope.launch {
-        loginState = try {
+        try {
+            val currState = _state.value
             val user = userRepository.getUser(sessionId).response!!
             val account = userRepository.getAccount(sessionId, user.id!!).response!!.accounts
             val transactions =
                 userRepository.getTransactionHistory(sessionId, user.id).response!!.transactions
             user.accounts = account
             user.transactions = transactions
-
             CurrentUser.user = user
-            LoginState.Success(user)
+
+            if (currState is State.Login) {
+                userPreferencesRepository.saveLoginInfo(sessionId, currState.password)
+                userPreferencesRepository.setIsLoggedIn(true)
+            }
+            val newState = State.Account(
+                user = user,
+                query = "",
+                accountFilter = AccountType.BRBS
+            )
+            _state.value = newState
         } catch (e: Exception) {
             Log.d("LOGIN", e.stackTraceToString())
-            LoginState.Error
+            val currState = _state.value
+            if (currState is State.Login) {
+                val newState = State.Login(
+                    netid = currState.netid,
+                    password = currState.password,
+                    failureMessage = e.stackTraceToString(),
+                    loading = false
+                )
+                _state.value = newState
+            }
+            userPreferencesRepository.saveLoginInfo("", "")
+            userPreferencesRepository.setIsLoggedIn(false)
         }
     }
-
-    fun logOut() = viewModelScope.launch {
-        CurrentUser.user = null
-        userPreferencesRepository.setIsLoggedIn(false)
-        userPreferencesRepository.saveLoginInfo("", "")
-    }
 }
 
 /**
```

Comment:
Delete `LoginState` and `LoggedInStatus` since these were artifacts from the previous messy implementation!